server/odc-service/src/main/java/com/oceanbase/odc/service/regulation/ruleset/RuleService.java

```diff
@@ -189,6 +191,34 @@ public Rule update(@NonNull Long rulesetId, @NonNull Long ruleId, @NonNull Rule
         return rule;
     }
 
+    @SkipAuthorize("internal usage")
+    @Transactional(rollbackFor = Exception.class)
+    public List<Rule> create(@NotNull Long rulesetId, @NotEmpty List<Rule> rules) {
+        List<RuleApplyingEntity> entities = rules.stream().map(rule -> {
+            RuleApplyingEntity entity = new RuleApplyingEntity();
+            entity.setOrganizationId(authenticationFacade.currentOrganizationId());
+            entity.setRulesetId(rulesetId);
+            entity.setRuleMetadataId(rule.getMetadata().getId());
+            entity.setLevel(rule.getLevel());
+            entity.setEnabled(rule.getEnabled());
+            entity.setPropertiesJson(JsonUtils.toJson(rule.getProperties()));
+            if (Objects.nonNull(rule.getAppliedDialectTypes())) {
+                entity.setAppliedDialectTypes(
+                        rule.getAppliedDialectTypes().stream().map(DialectType::name).collect(Collectors.toList()));
+            }
+            return entity;
+        }).collect(Collectors.toList());
+        ruleApplyingRepository.saveAll(entities);
```

Comment:
Better to use batchCreate for better performance, refer to `OdcJpaRepository`.

Reply:
Appreciate your reminder! I use `OdcJpaRepositiry#batchCreate` instead :)